command/kv_delete.go

```diff
@@ -0,0 +1,178 @@
+package command
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+)
+
+// KVDeleteCommand is a Command implementation that is used to delete a key or
+// prefix of keys from the key-value store.
+type KVDeleteCommand struct {
+	Ui cli.Ui
+}
+
+func (c *KVDeleteCommand) Help() string {
+	helpText := `
+Usage: consul kv delete [options] KEY_OR_PREFIX
+
+  Removes the value from Consul's key-value store at the given path. If no
+  key exists at the path, no action is taken.
+
+  To delete the value for the key named "foo" in the key-value store:
+
+      $ consul kv delete foo
+
+  To delete all keys which start with "foo", specify the -recurse option:
+
+      $ consul kv delete -recurse foo
+
+  This will delete the keys named "foo", "food", and "foo/bar/zip" if they
+  existed.
+
+` + apiOptsText + `
+
+KV Delete Options:
+
+  -cas                    Perform a Check-And-Set operation. If this value is
+                          specified without -modify-index, the key will first be
+                          fetched and the resulting ModifyIndex will be used on
+                          the next query. The default value is false.
+
+  -modify-index=<int>     Unsigned integer representing the ModifyIndex of the
+                          key. This is often combined with the -cas flag, but it
+                          can be specified for any key. The default value is 0.
+
+  -recurse                Recursively delete all keys with the path. The default
+                          value is false.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *KVDeleteCommand) Run(args []string) int {
+	cmdFlags := flag.NewFlagSet("get", flag.ContinueOnError)
+	cmdFlags.Usage = func() { c.Ui.Output(c.Help()) }
+	datacenter := cmdFlags.String("datacenter", "", "")
+	token := cmdFlags.String("token", "", "")
+	stale := cmdFlags.Bool("stale", false, "")
+	cas := cmdFlags.Bool("cas", false, "")
+	modifyIndex := cmdFlags.Uint64("modify-index", 0, "")
+	recurse := cmdFlags.Bool("recurse", false, "")
+	httpAddr := HTTPAddrFlag(cmdFlags)
+	if err := cmdFlags.Parse(args); err != nil {
+		return 1
+	}
+
+	key := ""
+
+	// Check for arg validation
+	args = cmdFlags.Args()
+	switch len(args) {
+	case 0:
+		key = ""
+	case 1:
+		key = args[0]
+	default:
+		c.Ui.Error(fmt.Sprintf("Too many arguments (expected 1, got %d)", len(args)))
+		return 1
+	}
+
+	// This is just a "nice" thing to do. Since pairs cannot start with a /, but
+	// users will likely put "/" or "/foo", lets go ahead and strip that for them
+	// here.
+	if len(key) > 0 && key[0] == '/' {
+		key = key[1:]
+	}
+
+	// If the key is empty and we are not doing a recursive delete, this is an
```

Comment:
This is a good point. I'm leaning towards "no" for this initial pass, since ideally you are using ACLs which prevent this from happening and/or backing up your Consul data.